web/packages/teleterm/src/ui/ModalsHost/modals/ReAuthenticate/ReAuthenticate.tsx

```diff
@@ -43,39 +43,34 @@ import { FieldSelect } from 'shared/components/FieldSelect';
 
 import { Option } from 'shared/components/Select';
 
-import { assertUnreachable } from 'shared/utils/assertUnreachable';
-
 import { useAppContext } from 'teleterm/ui/appContextProvider';
 import { LinearProgress } from 'teleterm/ui/components/LinearProgress';
 import svgHardwareKey from 'teleterm/ui/ClusterConnect/ClusterLogin/FormLogin/PromptPasswordless/hardware.svg';
-import { useLogger } from 'teleterm/ui/hooks/useLogger';
 import { routing } from 'teleterm/ui/uri';
-
-type MfaType = 'webauthn' | 'totp';
+import PromptSsoStatus from 'teleterm/ui/ClusterConnect/ClusterLogin/FormLogin/PromptSsoStatus';
 
 export const ReAuthenticate: FC<{
   promptMfaRequest: PromptMFARequest;
   onCancel: () => void;
   onSuccess: (otp: string) => void;
+  onSsoContinue: (redirectUrl: string) => void;
 }> = props => {
-  const logger = useLogger('ReAuthenticate');
   const { promptMfaRequest: req } = props;
 
-  const availableMfaTypes: MfaType[] = [];
-  // Add Webauthn first to prioritize it if both Webauthn and TOTP are available.
-  if (req.webauthn) {
-    availableMfaTypes.push('webauthn');
-  }
-  if (req.totp) {
-    availableMfaTypes.push('totp');
-  }
-  if (availableMfaTypes.length === 0) {
-    // This shouldn't happen but is technically allowed by the req data structure.
-    logger.warn('availableMfaTypes is empty, defaulting to webauthn and totp');
-    availableMfaTypes.push('webauthn', 'totp');
-  }
+  const availableMfaTypes = makeAvailableMfaTypes(req);
+
+  const [selectedMfaType, setSelectedMfaType] = useState<AvailableMfaType>(
+    availableMfaTypes[0]
+  );
+
+  useEffect(() => {
+    // If SSO is the default value, open the redirect window
+    // instead of waiting for the user to select SSO.
+    if (selectedMfaType.value === 'sso') {
```

Comment:
I agree, but at the same time the percent of users who are going to have both SSO and OTP enabled at the same time is probably going to be extremely small. SSO as an MFA option is going to be enabled only for SSO users. Those users at the moment are not even required to set up an MFA option. Requiring an extra click would be annoying to the (probably) vast majority of people who are going to use SSO MFA as the default option.

If anything, we could make it so that if `totp` is always in the array before `sso`, similar to how `webauthn` is before `sso`.

Reply:
Good idea, changing the options order should be good enough (I'd only leave a comment why we do it).